source/common/common/key_value_store_base.cc

```diff
@@ -59,43 +60,46 @@ bool KeyValueStoreBase::parseContents(absl::string_view contents,
       ENVOY_LOG(warn, error);
       return false;
     }
-    store.emplace(std::string(key.value()), std::string(value.value()));
+    store.insert(std::pair(std::string(key.value()), std::string(value.value())));
```

Comment:
Can the `contents` contain multiple keys with the same name (but different values)? i.e., should `insert_or_assign()` be used here?

Reply:
there shouldn't be - we could ASSERT or log warning if it exists but this is to load a prior serialized cache, and so should have unique keys.